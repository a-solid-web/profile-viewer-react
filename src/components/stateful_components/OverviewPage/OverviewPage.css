:root {
  background-color: #f9fafb;
}

.grid-container {
  background-color: #f9fafb;
  display: grid;
  grid-gap: 0px 25px;
  grid-template-columns: 1fr 2fr 1fr;
  grid-template-rows: 3em 3em 1fr;

  padding-left: 50px;
  padding-right: 50px;
}

#toggle {
  grid-column: 2/3;
  grid-row: 2/3;

  display: flex;
  justify-content: space-around;
  background-color: #ffff;
}

#toggle > div:first-of-type {
  color: #3256d7;
  display: flex;
  justify-content: center;
  align-items: center;
  border-bottom: 3px solid #3256d7;
  width: 100%;
}

#toggle > div:last-of-type {
  color: #c2c2c2;
  display: flex;
  justify-content: center;
  align-items: center;
  border-bottom: 3px solid #f6f6f6;
  width: 100%;
}

.requestcards {
  grid-column: 2/3;
  grid-row: 3;

  display: flex;
  flex-flow: column nowrap;
  align-content: space-around;
}

.requestcards-card {
  margin-top: 40px;
}

.requestcards-avatar {
  margin-right: 11px;
}

.requestcards-card-request {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;

  background-color: #ffff;
  padding: 20px;
  margin-top: 14px;
  box-shadow: 0px 6px 26px rgba(0, 0, 0, 0.02);
}

.requestcards-card-button {
  margin: 2px;
}
