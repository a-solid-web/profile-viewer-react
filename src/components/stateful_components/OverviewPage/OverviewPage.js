import React from "react";
import "./OverviewPage.css";
import { Button } from "yoda-design-system";

class OverviewPage extends React.Component {
  getRequests() {
    if (!this.props.requests) {
      return (
        <div className="requestcards-card-request">
          Looks like you don't have any requests at the moment
        </div>
      );
    } else {
      const requests = this.props.requests;
      return requests.map((item, i) => {
        return (
          <div className="requestcards-card" key={i}>
            <img
              src="https://via.placeholder.com/40?text=profile+picture"
              alt="avatar"
            />
            <strong>{item.name}</strong> wants to
            {item.requests.map((request, j) => {
              return (
                <div className="requestcards-card-request" key={j}>
                  {request}
                  <div>
                    <Button variant="outlined">Accept</Button>
                    <Button variant="outlined">Deny</Button>
                  </div>
                </div>
              );
            })}
          </div>
        );
      });
    }
  }

  render() {
    const requests = this.getRequests();
    return (
      <div className="grid-container">
        <div id="toggle">
          <div>Requests</div>
          <div>Activity</div>
        </div>
        <div className="requestcards">{requests}</div>
      </div>
    );
  }
}

export default OverviewPage;
