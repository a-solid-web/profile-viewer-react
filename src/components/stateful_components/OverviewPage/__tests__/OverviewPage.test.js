import React from "react";
import OverviewPage from "../OverviewPage";
import renderer from "react-test-renderer";

test("Request cards are displayed properly", () => {
  const component = renderer.create(
    <OverviewPage
      requests={Array(12).fill({
        name: "Malte Sielski",
        requests: ["View your phone number", "Edit your Birthdate"]
      })}
    />
  );

  let page = component.toJSON();
  expect(page).toMatchSnapshot();
});
