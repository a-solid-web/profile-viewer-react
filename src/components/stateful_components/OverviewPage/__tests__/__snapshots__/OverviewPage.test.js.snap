// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Request cards are displayed properly 1`] = `
<div
  className="grid-container"
>
  <div
    id="toggle"
  >
    <div>
      Requests
    </div>
    <div>
      Activity
    </div>
  </div>
  <div
    className="requestcards"
  >
    <div
      className="requestcards-card"
    >
      <img
        alt="avatar"
        src="https://via.placeholder.com/40?text=profile+picture"
      />
      <strong>
        Malte Sielski
      </strong>
       wants to
      <div
        className="requestcards-card-request"
      >
        View your phone number
        <div>
          <button
            className="Button-module_button__1-3q_ Button-module_button-outlined__22h7J "
            type="button"
          >
            Accept
          </button>
          <button
            className="Button-module_button__1-3q_ Button-module_button-outlined__22h7J "
            type="button"
          >
            Deny
          </button>
        </div>
      </div>
      <div
        className="requestcards-card-request"
      >
        Edit your Birthdate
        <div>
          <button
            className="Button-module_button__1-3q_ Button-module_button-outlined__22h7J "
            type="button"
          >
            Accept
          </button>
          <button
            className="Button-module_button__1-3q_ Button-module_button-outlined__22h7J "
            type="button"
          >
            Deny
          </button>
        </div>
      </div>
    </div>
    <div
      className="requestcards-card"
    >
      <img
        alt="avatar"
        src="https://via.placeholder.com/40?text=profile+picture"
      />
      <strong>
        Malte Sielski
      </strong>
       wants to
      <div
        className="requestcards-card-request"
      >
        View your phone number
        <div>
          <button
            className="Button-module_button__1-3q_ Button-module_button-outlined__22h7J "
            type="button"
          >
            Accept
          </button>
          <button
            className="Button-module_button__1-3q_ Button-module_button-outlined__22h7J "
            type="button"
          >
            Deny
          </button>
        </div>
      </div>
      <div
        className="requestcards-card-request"
      >
        Edit your Birthdate
        <div>
          <button
            className="Button-module_button__1-3q_ Button-module_button-outlined__22h7J "
            type="button"
          >
            Accept
          </button>
          <button
            className="Button-module_button__1-3q_ Button-module_button-outlined__22h7J "
            type="button"
          >
            Deny
          </button>
        </div>
      </div>
    </div>
    <div
      className="requestcards-card"
    >
      <img
        alt="avatar"
        src="https://via.placeholder.com/40?text=profile+picture"
      />
      <strong>
        Malte Sielski
      </strong>
       wants to
      <div
        className="requestcards-card-request"
      >
        View your phone number
        <div>
          <button
            className="Button-module_button__1-3q_ Button-module_button-outlined__22h7J "
            type="button"
          >
            Accept
          </button>
          <button
            className="Button-module_button__1-3q_ Button-module_button-outlined__22h7J "
            type="button"
          >
            Deny
          </button>
        </div>
      </div>
      <div
        className="requestcards-card-request"
      >
        Edit your Birthdate
        <div>
          <button
            className="Button-module_button__1-3q_ Button-module_button-outlined__22h7J "
            type="button"
          >
            Accept
          </button>
          <button
            className="Button-module_button__1-3q_ Button-module_button-outlined__22h7J "
            type="button"
          >
            Deny
          </button>
        </div>
      </div>
    </div>
    <div
      className="requestcards-card"
    >
      <img
        alt="avatar"
        src="https://via.placeholder.com/40?text=profile+picture"
      />
      <strong>
        Malte Sielski
      </strong>
       wants to
      <div
        className="requestcards-card-request"
      >
        View your phone number
        <div>
          <button
            className="Button-module_button__1-3q_ Button-module_button-outlined__22h7J "
            type="button"
          >
            Accept
          </button>
          <button
            className="Button-module_button__1-3q_ Button-module_button-outlined__22h7J "
            type="button"
          >
            Deny
          </button>
        </div>
      </div>
      <div
        className="requestcards-card-request"
      >
        Edit your Birthdate
        <div>
          <button
            className="Button-module_button__1-3q_ Button-module_button-outlined__22h7J "
            type="button"
          >
            Accept
          </button>
          <button
            className="Button-module_button__1-3q_ Button-module_button-outlined__22h7J "
            type="button"
          >
            Deny
          </button>
        </div>
      </div>
    </div>
    <div
      className="requestcards-card"
    >
      <img
        alt="avatar"
        src="https://via.placeholder.com/40?text=profile+picture"
      />
      <strong>
        Malte Sielski
      </strong>
       wants to
      <div
        className="requestcards-card-request"
      >
        View your phone number
        <div>
          <button
            className="Button-module_button__1-3q_ Button-module_button-outlined__22h7J "
            type="button"
          >
            Accept
          </button>
          <button
            className="Button-module_button__1-3q_ Button-module_button-outlined__22h7J "
            type="button"
          >
            Deny
          </button>
        </div>
      </div>
      <div
        className="requestcards-card-request"
      >
        Edit your Birthdate
        <div>
          <button
            className="Button-module_button__1-3q_ Button-module_button-outlined__22h7J "
            type="button"
          >
            Accept
          </button>
          <button
            className="Button-module_button__1-3q_ Button-module_button-outlined__22h7J "
            type="button"
          >
            Deny
          </button>
        </div>
      </div>
    </div>
    <div
      className="requestcards-card"
    >
      <img
        alt="avatar"
        src="https://via.placeholder.com/40?text=profile+picture"
      />
      <strong>
        Malte Sielski
      </strong>
       wants to
      <div
        className="requestcards-card-request"
      >
        View your phone number
        <div>
          <button
            className="Button-module_button__1-3q_ Button-module_button-outlined__22h7J "
            type="button"
          >
            Accept
          </button>
          <button
            className="Button-module_button__1-3q_ Button-module_button-outlined__22h7J "
            type="button"
          >
            Deny
          </button>
        </div>
      </div>
      <div
        className="requestcards-card-request"
      >
        Edit your Birthdate
        <div>
          <button
            className="Button-module_button__1-3q_ Button-module_button-outlined__22h7J "
            type="button"
          >
            Accept
          </button>
          <button
            className="Button-module_button__1-3q_ Button-module_button-outlined__22h7J "
            type="button"
          >
            Deny
          </button>
        </div>
      </div>
    </div>
    <div
      className="requestcards-card"
    >
      <img
        alt="avatar"
        src="https://via.placeholder.com/40?text=profile+picture"
      />
      <strong>
        Malte Sielski
      </strong>
       wants to
      <div
        className="requestcards-card-request"
      >
        View your phone number
        <div>
          <button
            className="Button-module_button__1-3q_ Button-module_button-outlined__22h7J "
            type="button"
          >
            Accept
          </button>
          <button
            className="Button-module_button__1-3q_ Button-module_button-outlined__22h7J "
            type="button"
          >
            Deny
          </button>
        </div>
      </div>
      <div
        className="requestcards-card-request"
      >
        Edit your Birthdate
        <div>
          <button
            className="Button-module_button__1-3q_ Button-module_button-outlined__22h7J "
            type="button"
          >
            Accept
          </button>
          <button
            className="Button-module_button__1-3q_ Button-module_button-outlined__22h7J "
            type="button"
          >
            Deny
          </button>
        </div>
      </div>
    </div>
    <div
      className="requestcards-card"
    >
      <img
        alt="avatar"
        src="https://via.placeholder.com/40?text=profile+picture"
      />
      <strong>
        Malte Sielski
      </strong>
       wants to
      <div
        className="requestcards-card-request"
      >
        View your phone number
        <div>
          <button
            className="Button-module_button__1-3q_ Button-module_button-outlined__22h7J "
            type="button"
          >
            Accept
          </button>
          <button
            className="Button-module_button__1-3q_ Button-module_button-outlined__22h7J "
            type="button"
          >
            Deny
          </button>
        </div>
      </div>
      <div
        className="requestcards-card-request"
      >
        Edit your Birthdate
        <div>
          <button
            className="Button-module_button__1-3q_ Button-module_button-outlined__22h7J "
            type="button"
          >
            Accept
          </button>
          <button
            className="Button-module_button__1-3q_ Button-module_button-outlined__22h7J "
            type="button"
          >
            Deny
          </button>
        </div>
      </div>
    </div>
    <div
      className="requestcards-card"
    >
      <img
        alt="avatar"
        src="https://via.placeholder.com/40?text=profile+picture"
      />
      <strong>
        Malte Sielski
      </strong>
       wants to
      <div
        className="requestcards-card-request"
      >
        View your phone number
        <div>
          <button
            className="Button-module_button__1-3q_ Button-module_button-outlined__22h7J "
            type="button"
          >
            Accept
          </button>
          <button
            className="Button-module_button__1-3q_ Button-module_button-outlined__22h7J "
            type="button"
          >
            Deny
          </button>
        </div>
      </div>
      <div
        className="requestcards-card-request"
      >
        Edit your Birthdate
        <div>
          <button
            className="Button-module_button__1-3q_ Button-module_button-outlined__22h7J "
            type="button"
          >
            Accept
          </button>
          <button
            className="Button-module_button__1-3q_ Button-module_button-outlined__22h7J "
            type="button"
          >
            Deny
          </button>
        </div>
      </div>
    </div>
    <div
      className="requestcards-card"
    >
      <img
        alt="avatar"
        src="https://via.placeholder.com/40?text=profile+picture"
      />
      <strong>
        Malte Sielski
      </strong>
       wants to
      <div
        className="requestcards-card-request"
      >
        View your phone number
        <div>
          <button
            className="Button-module_button__1-3q_ Button-module_button-outlined__22h7J "
            type="button"
          >
            Accept
          </button>
          <button
            className="Button-module_button__1-3q_ Button-module_button-outlined__22h7J "
            type="button"
          >
            Deny
          </button>
        </div>
      </div>
      <div
        className="requestcards-card-request"
      >
        Edit your Birthdate
        <div>
          <button
            className="Button-module_button__1-3q_ Button-module_button-outlined__22h7J "
            type="button"
          >
            Accept
          </button>
          <button
            className="Button-module_button__1-3q_ Button-module_button-outlined__22h7J "
            type="button"
          >
            Deny
          </button>
        </div>
      </div>
    </div>
    <div
      className="requestcards-card"
    >
      <img
        alt="avatar"
        src="https://via.placeholder.com/40?text=profile+picture"
      />
      <strong>
        Malte Sielski
      </strong>
       wants to
      <div
        className="requestcards-card-request"
      >
        View your phone number
        <div>
          <button
            className="Button-module_button__1-3q_ Button-module_button-outlined__22h7J "
            type="button"
          >
            Accept
          </button>
          <button
            className="Button-module_button__1-3q_ Button-module_button-outlined__22h7J "
            type="button"
          >
            Deny
          </button>
        </div>
      </div>
      <div
        className="requestcards-card-request"
      >
        Edit your Birthdate
        <div>
          <button
            className="Button-module_button__1-3q_ Button-module_button-outlined__22h7J "
            type="button"
          >
            Accept
          </button>
          <button
            className="Button-module_button__1-3q_ Button-module_button-outlined__22h7J "
            type="button"
          >
            Deny
          </button>
        </div>
      </div>
    </div>
    <div
      className="requestcards-card"
    >
      <img
        alt="avatar"
        src="https://via.placeholder.com/40?text=profile+picture"
      />
      <strong>
        Malte Sielski
      </strong>
       wants to
      <div
        className="requestcards-card-request"
      >
        View your phone number
        <div>
          <button
            className="Button-module_button__1-3q_ Button-module_button-outlined__22h7J "
            type="button"
          >
            Accept
          </button>
          <button
            className="Button-module_button__1-3q_ Button-module_button-outlined__22h7J "
            type="button"
          >
            Deny
          </button>
        </div>
      </div>
      <div
        className="requestcards-card-request"
      >
        Edit your Birthdate
        <div>
          <button
            className="Button-module_button__1-3q_ Button-module_button-outlined__22h7J "
            type="button"
          >
            Accept
          </button>
          <button
            className="Button-module_button__1-3q_ Button-module_button-outlined__22h7J "
            type="button"
          >
            Deny
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
